Remote_Labs/Working-With-FPGAs.md

```diff
@@ -291,12 +291,374 @@ To exit the terminal emulation, hit `Control-A` and then `k`, then `y` to confir
+
+Building with petalinux is done on chococat. But the zcu102 is connected to keroppi. The reason for this is because a custom glibc had to be made on keroppi to fix a MATLAB bug. Simulink won't run without this patch from Mathworks. Unfortunately, it has side effects. One of those side effects is that petalinux won't run. There is probably a better way to compartmentalize this, but we haven't found it or done it yet. So, here is how to move the files built on chococat to the /tftpboot folder on keroppi:
+	
+```
+abraxas3d@chococat:/tftpboot$ nano TEST_FILE
+abraxas3d@chococat:/tftpboot$ ls
+abraxas3d  adiboot  salvy  TEST_FILE
+abraxas3d@chococat:/tftpboot$ 
+
+abraxas3d@keroppi:/tftpboot$ scp abraxas3d@chococat:/tftpboot/* .
+abraxas3d@chococat's password:
+scp: /tftpboot/abraxas3d: not a regular file
+scp: /tftpboot/adiboot: not a regular file
+scp: /tftpboot/salvy: not a regular file
+TEST_FILE                                                             100%   20    36.0KB/s   00:00
+abraxas3d@keroppi:/tftpboot$ ls
+abraxas3d  TEST_FILE
+```
+
+Make sure there's a hw_server process running.
+
+Monitor the serial console of the target. This may be /dev/ttyUSB1 or /dev/ttyUSB2. If you can't use either port, somebody else may have the console port open.
+```
+screen /dev/ttyUSB1 115200
+```
+
+Now go for the boot:
+```
+petalinux-package --prebuilt --fpga <location of bitfile> --force
+petalinux-boot --jtag --prebuilt 2 --hw_server-url TCP:keroppi:3121
+```
+
+If you get `Will not program bitstream on the target` then you might have forgotten to choose the `Include bitstream` option when you exported the XSA file.
+
+If you get a warning about your $XTERM and `Inappropriate ioctl for device.` you can ignore it.
+
+If you get a `Memory read error` OR `No supported FPGA device found` then type `reboot` at the serial console prompt and wait for it to finish. It won't actually reboot and return to the prompt. Then retry the `petalinux-boot` command.
+
+Watch the messages from petalinux-boot. Make sure it confirms that it is downloading the bitstream to the target.
+
+On the target's serial console, watch for the `Zynq>` prompt.
+There may be a bunch of messages claiming that the TFTP server died. Don't worry about that. Just wait for it to stop and display the `Zynq>` prompt.
+
+```
+Zynq> setenv serverip 10.73.1.94
+Zynq> setenv ipaddr 10.73.1.16 
+Zynq> pxe get
+Zynq> pxe boot
+```
+This should boot petalinux on the zc706.
+
+Note that the IP address set with setenv at the Zync> prompt does not stick. By default, the Linux system you built will use DHCP and get its own address. You can find out what address was assigned with `ip addr` on the serial console. If you followed the recommended procedure in petalinux-config above, it should already be set to `10.73.1.9`.
```

Comment:
Shouldn't this say "it should already be set to `10.73.1.16` here?

Reply:
Isn't the IP address for Chococat hardware 10.73.1.9?

Reply:
Oh, you're right. I assumed line 517 was right and the error was in line 523, but the opposite is true. I think line 516 is also wrong; .94 is keroppi and this is happening on chococat, which is .93